src/test/java/org/jenkinsci/plugins/github_branch_source/GithubSCMSourceTagsTest.java

```diff
@@ -0,0 +1,443 @@
+                        new GitHubTagSCMHead("existent-multiple-tags1", System.currentTimeMillis()),
+                        new GitHubTagSCMHead("existent-multiple-tags2", System.currentTimeMillis()))));
+        GitHubSCMSourceContext context = new GitHubSCMSourceContext(null, mockSCMHeadObserver);
+        context.wantTags(true);
+        GitHubSCMSourceRequest request = context.newRequest(revisions()[0], null);
+        GHRepository repoSpy = Mockito.spy(repo);
+        PagedIterable<GHRef> iterableSpy = Mockito.spy(repoSpy.listRefs("tags"));
+        Mockito.when(repoSpy.listRefs("tags")).thenReturn(iterableSpy);
+        Mockito.when(iterableSpy.iterator()).thenThrow(expectedError).thenThrow(expectedError2);
+
+        Iterator<GHRef> tags = new GitHubSCMSource.LazyTags(request, repoSpy).iterator();
+        assertEquals(Collections.<GHRef>emptyList().iterator(), tags);
+        try{
+            Iterator<GHRef> tags2 = new GitHubSCMSource.LazyTags(request, repoSpy).iterator();
+            fail("This should throw an exception");
+        }
+        catch(Error e){
+            assertEquals("Bad Tag Request IOError", e.getMessage());
+        }
+    }
+
+    @Test
+    public void testExistingMultipleTagsIteratorGHFileNotFoundExceptionOnHasNext() throws IOException {
+        SCMHeadObserver mockSCMHeadObserver = Mockito.mock(SCMHeadObserver.class);
+        Error expectedError = new Error("Bad Tag Request", new GHFileNotFoundException());
+        Error expectedError2 = new Error("Bad Tag Request IOError", new IOException());
+        Mockito.when(mockSCMHeadObserver.getIncludes()).thenReturn(
+                new HashSet<>(Arrays.asList(
+                        new GitHubTagSCMHead("existent-multiple-tags1", System.currentTimeMillis()),
+                        new GitHubTagSCMHead("existent-multiple-tags2", System.currentTimeMillis()))));
+        GitHubSCMSourceContext context = new GitHubSCMSourceContext(null, mockSCMHeadObserver);
+        context.wantTags(true);
+        GitHubSCMSourceRequest request = context.newRequest(revisions()[0], null);
+        GHRepository repoSpy = Mockito.spy(repo);
+        PagedIterable<GHRef> iterableSpy = Mockito.spy(repoSpy.listRefs("tags"));
+        Mockito.when(repoSpy.listRefs("tags")).thenReturn(iterableSpy);
+
+        PagedIterator<GHRef> iteratorSpy = Mockito.spy(iterableSpy.iterator());
+        Mockito.when(iterableSpy.iterator()).thenReturn(iteratorSpy);
+        Mockito.when(iteratorSpy.hasNext()).thenThrow(expectedError);//.thenThrow(expectedError2);
+        Iterator<GHRef>  tags = new GitHubSCMSource.LazyTags(request, repoSpy).iterator();
+        assertFalse(tags.hasNext());
+    }
+    @Test
+    public void testExistingMultipleTagsIteratorIOErrorOnHasNext() throws IOException {
+        SCMHeadObserver mockSCMHeadObserver = Mockito.mock(SCMHeadObserver.class);
+        Error expectedError = new Error("Bad Tag Request", new IOException());
```

Comment:
Comment that the `IOException` here has a different code path that `FileNotFoundException`